lib/src/list_mutations.dart

```diff
@@ -117,7 +117,7 @@ SourceEdit _appendToBlockList(
   // Adjusts offset to after the trailing newline of the last entry, if it exists
   if (list.isNotEmpty) {
     final lastValueSpanEnd = list.nodes.last.span.end.offset;
-    final nextNewLineIndex = yaml.indexOf('\n', lastValueSpanEnd);
+    final nextNewLineIndex = yaml.indexOf('\n', lastValueSpanEnd-1);
```

Comment:
make sure your editor runs `dart format` on save, it'll make your life a lot easier :D